homeassistant/components/uvc/camera.py

```diff
@@ -97,14 +103,31 @@ def name(self):
     @property
     def is_recording(self):
         """Return true if the camera is recording."""
-        caminfo = self._nvr.get_camera(self._uuid)
-        return caminfo['recordingSettings']['fullTimeRecordEnabled']
+        return self._caminfo['recordingSettings']['fullTimeRecordEnabled']
 
     @property
     def motion_detection_enabled(self):
         """Camera Motion Detection Status."""
-        caminfo = self._nvr.get_camera(self._uuid)
-        return caminfo['recordingSettings']['motionRecordEnabled']
+        return self._caminfo['recordingSettings']['motionRecordEnabled']
+
+    @property
+    def supported_features(self):
+        """Return supported features."""
+        for channel in self._caminfo['channels']:
+            if channel['isRtspEnabled']:
+                return SUPPORT_STREAM
+        return 0
+
+    @property
+    def stream_source(self):
+        """Return the stream source."""
+        for channel in self._caminfo['channels']:
+            if channel['isRtspEnabled']:
```

Comment:
It would make more sense to calculate the stream source URL and supported_features inside the `camera_info` method.

Reply:
And if you do, rename it to `load_camera_info` to make it clear that the function has side effects.

Reply:
to do it this way, do I add something like `self._isRtspEnabled`, and `self._rtspUri`in `__init__` and populate them in `load_camera_info`?

do I just do `self._rtspUri` and only send stream source and supported_features if `self._rtspUri` is not None?

Reply:
yeah, that makes sense.

Reply:
👍